relational_operators/HashJoinOperator.hpp

```diff
@@ -141,6 +141,12 @@ class HashJoinOperator : public RelationalOperator {
     return output_relation_.getID();
   }
 
+  void doneFeedingInputBlocks(const relation_id rel_id) override {
```

Comment:
Not exactly. This fix is relevant when there's a HashJoin is a consumer of the output of some dependent operator (most likely a select).  The Foreman needs to notify HashJoin::doneFeedingInputBlocks only when the select stops pipelining blocks. 

As you know that BuildHash is also a dependent operator of HashJoin, the HashJoin is also notified of doneFeedingInputBlocks when BuildHash is completed. But HashJoin should ignore that message.